vertical-pod-autoscaler/recommender/util/histogram_options.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"errors"
+	"math"
+)
+
+// HistogramOptions define the number and size of buckets of a histogram.
+type HistogramOptions interface {
+	NumBuckets() int
+	// Returns the index of the bucket to which the given value falls.
+	// If the value is outside of the range covered by the histogram, it
+	// returns the closest bucket (either the first or the last one).
+	FindBucket(value float64) int
+	// Returns the start of the bucket with a given index. If the index is
+	// outside the [0..NumBuckets() - 1] range, the result is undefined.
+	GetBucketStart(bucket int) float64
+}
+
+// NewLinearHistogramOptions returns HistogramOptions describing a histogram
+// with a given number of fixed-size buckets, with the first bucket starting
+// at 0.0. Requires maxValue > 0, bucketSize > 0.
+func NewLinearHistogramOptions(
+	maxValue float64, bucketSize float64) (HistogramOptions, error) {
```

Comment:
I don't see a use case for minValue != 0 at this moment. We're going to use histograms to measure some "physical quantities" which always range from 0 to some maximum.